### PR TITLE
[MRG+1] Custom pivot for barbs

### PR DESCRIPTION
**Pull request by dopplershift**
This expands the `pivot` kwarg to take a float value, allowing the user to manually specify where the barb should start. This allows, for instance, the barbs to be offset outward from the center plot, freeing the center for other plot elements.

In meteorology, this is useful for standard plots where sky cover should be at the center, not the end of the wind barb.

### CODE REVIEW

**Comment by mdboom** [comment]
A picture example?

**Comment by dopplershift** [comment]
Like this:
image

**Comment by dopplershift** [comment]
Rebased and added a test for the new pivot.

**Comment by QuLogic** [comment]
It appears that your email address changed... Anyway, the test is failing either way.

**Comment by dopplershift** [comment]
Great, a 2 pixel difference (the end of one of the barbs), at least on windows and Python 2.7. I just can't wait to figure this one out.

**Comment by dopplershift** [comment]
Fixed the email. The test failure was two pixels at the end of one of the barbs, only on windows/linux. I tweaked the test to just hopefully avoid it. (I started digging in, but the point locations generated across the versions were identical, which means the difference was somewhere in the drawing code...)

**Comment by dopplershift** [comment]
Ok, everything is green! :shipit:

File: lib/matplotlib/quiver.py [hunk 1]
```diff
@@ -1009,7 +1009,8 @@ def _make_barbs(self, u, v, nflags, nbarbs, half_barb, empty_flag, length,
 
         *pivot* specifies the point on the barb around which the
         entire barb should be rotated.  Right now, valid options are
-        'head' and 'middle'.
+        'tip' and 'middle'. Can also be a number, which shifts the start
+        of the barb that many units from the origin.
```
Comment (QuLogic) [inline lib/matplotlib/quiver.py]: But what "units" is it? Pixels?

Comment (dopplershift) [inline lib/matplotlib/quiver.py]: What units are polygons/paths/markers normally in? I don't think it's pixels, but it's not data coordinates either.

Comment (QuLogic) [inline lib/matplotlib/quiver.py]: I'd have thought points, but the scatter size is defined in points. I don't remember the scaling, but if it's a square, then the radius is √49/2 = 3.5. If it's actual circular area, it's √(49/π) = 3.949. But neither of these seem to match the shift defined in the test of 1.7, so I'm still not sure.

Comment (QuLogic) [inline lib/matplotlib/quiver.py]: It looks like the default pivot uses [some factor of `length`](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/quiver.py#L1050) and `length` is [defined in points](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/quiver.py#L1008), so I guess this _should_ be points.

BTW, in looking through this, I found a typo in the docs; do you mind fixing it here?

This section says the [default length is 9](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/quiver.py#L783-L786], but the [default appears to be 7](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/quiver.py#L916).

**Comment by dopplershift** [comment]
Does that work for you @QuLogic ?

**Pull request merged**